### PR TITLE
Add tpu v4 to metadata and catalog

### PR DESCRIPTION
**Pull request by infwinston**
This PR adds tpu-v4 in our metadata csv and catalog. The info is based on GCP official [doc](https://cloud.google.com/tpu/pricing) and 
```
gcloud compute tpus accelerator-types list --zone us-central2-b
ACCELERATOR_TYPE
v4-1024
v4-1152
v4-128
v4-1280
v4-1408
v4-1536
...
```
For some reason, there are much more different config on v4 cores than v3.

I tested 
```
resources:
  accelerators: tpu-v4-8
  accelerator_args:
    runtime_version: tpu-vm-v4-base
    tpu_vm: True
```
with the mnist example in https://github.com/skypilot-org/skypilot/blob/master/examples/tpu/tpuvm_mnist.yaml
note that `runtime_version` needs to be `tpu-vm-v4-base` instead of `tpu-vm-base`.
and it works well
```
> sky launch examples/tpu/tpuvm_mnist.yaml
...
(tpuvm_mnist, pid=25184) I0309 21:44:10.261609 139933689456640 main.py:50] JAX process: 0 / 1
(tpuvm_mnist, pid=25184) I0309 21:44:10.261830 139933689456640 main.py:51] JAX local devices: [TpuDevice(id=0, process_index=0, coords=(0,0,0), core_on_chip=0), TpuDevice(id=1, process_inde
x=0, coords=(1,0,0), core_on_chip=0), TpuDevice(id=2, process_index=0, coords=(0,1,0), core_on_chip=0), TpuDevice(id=3, process_index=0, coords=(1,1,0), core_on_chip=0)]
...
(tpuvm_mnist, pid=25184) I0309 21:44:41.000832 139933689456640 train.py:146] epoch:  9, train_loss: 0.0116, train_accuracy: 99.63, test_loss: 0.0259, test_accuracy: 99.16
(tpuvm_mnist, pid=25184) I0309 21:44:42.356218 139933689456640 train.py:146] epoch: 10, train_loss: 0.0090, train_accuracy: 99.74, test_loss: 0.0264, test_accuracy: 99.07
```

### CODE REVIEW

File: metadata/tpu/pricing.csv [hunk 1]
```diff
@@ -1,28 +1,63 @@
...
-europe-west4,tpu-v3-2048,2048,614.4
-asia-east1,tpu-v2-8,5.22,1.566
+Region,TPU type,Price,Spot price
+us-central1,tpu-v2-8,4.5,1.35
+us-central1,tpu-v2-32,24,7.2
+us-central1,tpu-v2-128,96,28.8
+us-central1,tpu-v2-256,192,57.6
+us-central1,tpu-v2-512,384,115.2
+us-central1,tpu-v3-8,8,2.4
+us-central2,tpu-v4-8,12.88,3.88
```
Comment (WoosukKwon) [inline metadata/tpu/pricing.csv:8]: nit: How did you get the spot price? As the discount rate is 70%, it's `3.864` in my calculation.

**Review by WoosukKwon — APPROVED** [review]
LGTM. Thanks for add TPU v4!

File: metadata/tpu/pricing.csv (hunk 1 above)
Comment (infwinston) [reply metadata/tpu/pricing.csv:8]: I got it from here. it seems not to be exactly 70%..
image

Comment (infwinston) [reply metadata/tpu/pricing.csv:8]: Ah or you think `0.97` is rounded from `0.966`

Comment (infwinston) [reply metadata/tpu/pricing.csv:8]: let me update with `0.966` seems to be more reasonable.

Comment (WoosukKwon) [reply metadata/tpu/pricing.csv]: 👍

**Comment by infwinston** [comment]
Thanks a lot for the quick review @WoosukKwon !

**Comment by Michaelvll** [comment]
Can we include them in our fetcher as well, i.e. will our current `gcp_fetch` fetch the prices for tpu_v4?

**Comment by infwinston** [comment]
yea we should consider fetching TPU prices from API. we can try adding them in @WoosukKwon 's next PR for fixing GCP catalog fetcher.

**Comment by Michaelvll** [comment]
Seems this PR is causing the GitHub CI to fail for all the PRs: https://github.com/skypilot-org/skypilot/actions/runs/4381655786/jobs/7670111729. 

Seems we need to fix `sky.clouds.service_catalog.__init__.get_tpus()` and other places that we hardcoded the tpu types?

**Pull request merged**